Airtrail JSON import fails

### The bug

When I attempt to import airtrail.json from an Airtrail export, it fails with the following popup.

![image](https://github.com/user-attachments/assets/b2b5a0d3-2c0d-4adb-a611-1e0240769235)

This happens regardless of what state Airtrail is in. I have tried to import the contents immediately after export, and also into a fresh instance of Airtrail, both times I get the same error. I see no logs in the docker container. But I do see en error message when using the browser console.

### Operating System that is running AirTrail

Ubuntu

### AirTrail Version

1.1.0

### Your docker-compose.yml content

```YAML
services:
  db:
    image: postgres:16-alpine
    restart: always
    env_file:
      - .env
    environment:
      POSTGRES_DB: ${DB_DATABASE_NAME}
      POSTGRES_USER: ${DB_USERNAME}
      POSTGRES_PASSWORD: ${DB_PASSWORD}
    volumes:
      - db_data:/var/lib/postgresql/data
    healthcheck:
      test: ["CMD-SHELL", "pg_isready -U ${DB_USERNAME} -d ${DB_DATABASE_NAME}"]
      interval: 5s
      timeout: 5s
      retries: 5

  airtrail:
    image: johly/airtrail:latest
    restart: always
    env_file:
      - .env
    ports:
      - 1380:3000
    depends_on:
      db:
        condition: service_healthy

volumes:
  db_data:
```


### Your .env content

```Shell
# Your domain, e.g https://example.com
# You might have to add :443 if you are using https through a reverse proxy
ORIGIN=http://XXXXXXX:1380

# The database URL used by the application.
# If you are using the provided docker-compose file, you should only change the PASSWORD_HERE
# If you are using your own database, you should change this to the correct URL
DB_URL=postgres://airtrail:XXXXXXXXXX@db:5432/airtrail

# Values below this line are only for the default provided postgres database
###################################################################################
# Connection secret for postgres. You should change it to a random password
# Please use only the characters `A-Za-z0-9`, without special characters or spaces
# When you change the DB_PASSWORD, you should also update the DB_URL accordingly
DB_PASSWORD=XXXXXXXXXX

# The values below this line do not need to be changed
###################################################################################
DB_DATABASE_NAME=airtrail
DB_USERNAME=airtrail
```


### Reproduction steps

1. Export flights using the Export menu in settings (using JSON format)
2. Import the same exported file from the Import menu
3. Observe error popup, and below logs

This also works with a trimmed down export JSON, attached here: 
[airtrail.json](https://github.com/user-attachments/files/17720420/airtrail.json)



### Relevant log output

```shell
index.MfnEzb-6.js:22 Error: [
  {
    "code": "invalid_string",
    "validation": "datetime",
    "message": "Invalid datetime",
    "path": [
      "flights",
      0,
      "departure"
    ]
  },
  {
    "code": "invalid_string",
    "validation": "datetime",
    "message": "Invalid datetime",
    "path": [
      "flights",
      0,
      "arrival"
    ]
  }
]
    at vk (index.MfnEzb-6.js:20:20983)
    at bk (index.MfnEzb-6.js:20:22011)
    at async HTMLButtonElement.h [as __click] (index.MfnEzb-6.js:22:420)
```


### Additional information

Seems like the error is popping up [here](https://github.com/johanohly/AirTrail/blob/870cd6842a17e69dd03eef0be045775d9d5d80ef/src/lib/import/airtrail.ts#L16-L17). I don't know Typescript enough to even attempt to fix this.

--- a/src/lib/import/airtrail.ts
+++ b/src/lib/import/airtrail.ts
@@ -13,8 +13,14 @@ const AirTrailFile = z.object({
   flights: z
     .object({
       date: z.string().regex(/^\d{4}-\d{2}-\d{2}$/),
-      departure: z.string().datetime('Invalid datetime').nullable(),
-      arrival: z.string().datetime('Invalid datetime').nullable(),
+      departure: z
+        .string()
+        .datetime({ offset: true, message: 'Invalid datetime' })
+        .nullable(),
+      arrival: z
+        .string()
+        .datetime({ offset: true, message: 'Invalid datetime' })
+        .nullable(),
       duration: z.number().int().positive().nullable(),
     })
     .merge(flightAirportsSchema)
@@ -80,7 +86,18 @@ export const processAirTrailFile = async (input: string) => {
       const user = dataUser
         ? users.find((user) => user.username === dataUser?.username)
         : null;
-      const guestName = user ? null : seat.guestName;
+      /*
+      1. If the user is known, no guest name is needed.
+      2. If the user is unknown, but the guest name is known, use the guest name.
+      3. If the user is unknown and the guest name is unknown, use the provided display name (this could happen if the user is not in the database).
+       */
+      const guestName = user
+        ? null
+        : seat.guestName
+          ? seat.guestName
+          : dataUser
+            ? dataUser.displayName
+            : null;
 
       return {
         ...seat,

